plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -611,8 +611,13 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
             // Parallel read-only trx execution enabled.
             // Store the transaction in read-only-trx queue so that it is
             // executed in read window
-            std::lock_guard<std::mutex> g( _ro_trx_queue.mtx );
-            _ro_trx_queue.queue.push_back({trx, std::move(next)});
+            app().executor().post(priority::low, exec_queue::read_only_trx_safe, [this, trx=trx, next{std::move(next)}]() mutable {
+               auto retry = process_read_only_transaction( trx, next );
```

Comment:
Now a read-only transaction can be executed in either read window or write window. If executed in write window, in `push_read_only_transaction` called by `process_read_only_transaction`, `_read_window_deadline` is unlimited; if executed in read window, `_read_window_deadline`  is `fc::time_point::now() + _ro_read_window_effective_time_us`. Is this what we want?

Reply:
The read-only-trx will be limited by `max-transaction-time` then. Since this is not allowed on a block producer there is no block deadline to enforce. However, I do think it should be limited to `_ro_write_window_time_us` for the write window. I have made that change.